locations/spiders/kfc_za.py

```diff
@@ -0,0 +1,56 @@
+import scrapy
+from scrapy.http import JsonRequest
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+from locations.pipelines.address_clean_up import clean_address
+from locations.spiders.kfc_us import KFC_SHARED_ATTRIBUTES
+
+
+class KfcZASpider(scrapy.Spider):
+    name = "kfc_za"
+    item_attributes = KFC_SHARED_ATTRIBUTES
+    start_urls = ["https://orderserv-kfc-eu-olo-api.yum.com/dev/v1/stores/"]
+    tenant_id = "5f89a603781e4df48d31337a6c1252a8"
+    requires_proxy = True  # May require ZA proxy, possibly residential IPs only.
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield JsonRequest(url=url, headers={"x-tenant-id": self.tenant_id})
+
+    def parse(self, response):
+        for location in response.json():
+            item = DictParser.parse(location)
+            if location["code"] == "1000" or location["code"] == "1001":
+                # Ignore dummy stores used for internal testing/development
+                continue
+            item["ref"] = location["code"]
+            item["street_address"] = clean_address(location["localAddress"][0]["address"]["addressLines"])
+            item["city"] = location["localAddress"][0]["address"]["city"]
+            item["state"] = location["localAddress"][0]["address"]["state"]
+            item["country"] = location["localAddress"][0]["address"]["country"]
+            item["postcode"] = location["localAddress"][0]["address"]["pinCode"]
+            for contact_method in location["contacts"]:
+                if contact_method["key"] == "email":
+                    item["email"] = contact_method["value"]
+                elif contact_method["key"] == "phoneNumber":
+                    item["phone"] = contact_method["value"]
+            web_path = item["name"].lower().replace(" ", "-") + "/" + item["postcode"]
+            item["website"] = "https://order.kfc.com.za/restaurants/" + web_path
+            details_url = "https://orderserv-kfc-apac-olo-api.yum.com/dev/v1/stores/details/" + web_path
```

Comment:
I'm guessing the domain here should be `orderserv-kfc-eu-olo-api.yum.com`

Reply:
Thank, fixed now